src/main/java/org/opensearch/timeseries/util/RestHandlerUtils.java

```diff
@@ -130,18 +135,18 @@ public static XContentParser createXContentParserFromRegistry(NamedXContentRegis
     }
 
     /**
-     * Check if there is configuration/syntax error in feature definition of anomalyDetector
-     * @param anomalyDetector detector to check
-     * @param maxAnomalyFeatures max allowed feature number
+     * Check if there is configuration/syntax error in feature definition of config
+     * @param config config to check
+     * @param maxFeatures max allowed feature number
      * @return error message if error exists; otherwise, null is returned
      */
-    public static String checkAnomalyDetectorFeaturesSyntax(AnomalyDetector anomalyDetector, int maxAnomalyFeatures) {
-        List<Feature> features = anomalyDetector.getFeatureAttributes();
+    public static String checkFeaturesSyntax(Config config, int maxFeatures) {
+        List<Feature> features = config.getFeatureAttributes();
         if (features != null) {
-            if (features.size() > maxAnomalyFeatures) {
-                return "Can't create more than " + maxAnomalyFeatures + " anomaly features";
+            if (features.size() > maxFeatures) {
+                return "Can't create more than " + maxFeatures + " features";
             }
-            return validateFeaturesConfig(anomalyDetector.getFeatureAttributes());
+            return validateFeaturesConfig(config.getFeatureAttributes());
         }
         return null;
     }
```

Comment:
Yes. Changed comment on line 200 like

```
/**
     * Wrap action listener to avoid return verbose error message and wrong 500 error to user.
     * Suggestion for exception handling in timeseries analysis (e.g., AD and Forecast):
     * 1. If the error is caused by wrong input, throw IllegalArgumentException exception.
     * 2. For other errors, please use TimeSeriesException or its subclass, or use
     *    OpenSearchStatusException.
     *
     * TODO: tune this function for wrapped exception, return root exception error message
     *
     * @param actionListener action listener
     * @param generalErrorMessage general error message
     * @param <T> action listener response type
     * @return wrapped action listener
     */
    public static <T> ActionListener wrapRestActionListener(ActionListener<T> actionListener, String generalErrorMessage) {
        return ActionListener.<T>wrap(r -> { actionListener.onResponse(r); }, e -> {
            logger.error("Wrap exception before sending back to user", e);
            Throwable cause = Throwables.getRootCause(e);
            if (isProperExceptionToReturn(e)) {
                actionListener.onFailure(e);
            } else if (isProperExceptionToReturn(cause)) {
                actionListener.onFailure((Exception) cause);
            } else {
                RestStatus status = isBadRequest(e) ? BAD_REQUEST : INTERNAL_SERVER_ERROR;
                String errorMessage = generalErrorMessage;
                if (isBadRequest(e) || e instanceof TimeSeriesException) {
                    errorMessage = e.getMessage();
                } else if (cause != null && (isBadRequest(cause) || cause instanceof TimeSeriesException)) {
                    errorMessage = cause.getMessage();
                }
                actionListener.onFailure(new OpenSearchStatusException(errorMessage, status));
            }
        });
    }
```